nomad/structs/structs.go

```diff
@@ -9723,6 +9729,22 @@ func (a *Allocation) WaitClientStop() time.Time {
 	return t.Add(*tg.StopAfterClientDisconnect + kill)
 }
 
+// DisconnectTimeout uses the MaxClientDisconnect to compute when the allocation
+// should transition to lost.
+func (a *Allocation) DisconnectTimeout(now time.Time) time.Time {
+	tg := a.Job.LookupTaskGroup(a.TaskGroup)
```

Comment:
Can `a.Job` ever be nil? I know we set it that way in some circumstances.

Reply:
Updated in next push.